src/modules/pm.c

```diff
@@ -142,26 +144,42 @@ static void publish_susp_req(const bool new) {
     M_PUB(suspend_req);
 }
 
-static void on_pm_req(const bool new) {
-    if (new && pm_inh_token == -1) {
-        USERBUS_ARG_REPLY(pm_args, parse_bus_reply, &pm_inh_token,
-                            "org.freedesktop.PowerManagement.Inhibit",
-                            "/org/freedesktop/PowerManagement/Inhibit",
-                            "org.freedesktop.PowerManagement.Inhibit",
-                            "Inhibit");
-        if (call(&pm_args, "ss", "Clight", "PM inhibition.") == 0) {
-            DEBUG("Holding PowerManagement inhibition.\n");
+static bool acquire_lock(void) {
+    if (pm_inh_token == -1) {
+        SYSBUS_ARG_REPLY(pm_args, parse_bus_reply, &pm_inh_token,
+                         "org.freedesktop.login1", "/org/freedesktop/login1",
+                         "org.freedesktop.login1.Manager", "Inhibit");
+        int ret = call(&pm_args, "ssss", "idle", "Clight", "Idle inhibitor.",
+                       "block");
+        if (ret < 0) {
+            DEBUG("Failed to parse D-Bus response for idle inhibit\n");
+        } else if (pm_inh_token < 0) {
+            DEBUG("Failed to copy lock file\n");
+        } else {
+            return true;
+        }
+    }
+    return false;
+}
+
+static bool release_lock(void) {
+    if (pm_inh_token >= 0) {
+        (close(pm_inh_token));
+        pm_inh_token = -1;
+        return true;
+    }
+    return false;
+}
+
+static void on_pm_req(const bool new, const bool old) {
+    if (new && !old) {
+        if (acquire_lock()) {
```

Comment:
I really like the split `acquire_lock` and `release_lock`.
I think we could support both `/org/freedesktop/PowerManagement/Inhibit`, therefore just adding new:
* `acquire_logind_lock`
* `acquire_freedesktop_lock`

and their `release_foo_lock`.
WDYT?

Supporting both should work on more systems (ie: non systemd ones!).